baselines/asset/src/v1/asset_service_client.ts.baseline

```diff
@@ -92,14 +157,16 @@ export class AssetServiceClient {
   constructor(opts?: ClientOptions) {
     // Ensure that options include all the required fields.
     const staticMembers = this.constructor as typeof AssetServiceClient;
-    const servicePath = opts?.servicePath || opts?.apiEndpoint || staticMembers.servicePath;
+    // See: https://google.aip.dev/auth/4114
+    const [cert, key] = detectClientCertificate(opts);
```

Comment:
I'm a little bit concerned about using synchronous file API in the constructor. We have an asynchronous `initialize()` method, maybe this logic can be moved there (using promisified version of `fs`)?

Reply:
I've refactored to use async methods.